lib/assets/chart_cell/main.js

```diff
@@ -147,40 +173,56 @@ export function init(ctx, payload) {
 
     data() {
       return {
-        fields: payload.fields,
+        rootFields: payload.root_fields,
+        layers: payload.layers,
         dataOptions: payload.data_options,
         missingDep: payload.missing_dep,
-        chartOptions: ["point", "bar", "line", "area", "boxplot"],
-        typeOptions: ["quantitative", "nominal", "ordinal"],
+        chartOptions: ["point", "bar", "line", "area", "boxplot", "rule"],
+        typeOptions: ["quantitative", "nominal", "ordinal", "temporal"],
         aggregateOptions: ["sum", "mean"],
-        dataVariables: payload.data_options.map(data => data["variable"]),
+        dataVariables: payload.data_options.map((data) => data["variable"]),
       };
     },
 
     computed: {
-      axisOptions() {
-        const dataVariable = this.fields.data_variable;
-        const dataOptions = this.dataOptions.find(data => data["variable"] === dataVariable);
-        return dataOptions ? dataOptions["columns"].concat("__count__") : [];
-      },
       noDataVariable() {
-        return !this.fields.data_variable;
+        return !this.layers[0].data_variable;
       },
-      noColorField() {
-        return !this.fields.color_field || this.fields.color_field === "__count__";
-      },
-      noYField() {
-        return !this.fields.y_field || this.fields.y_field === "__count__";
-      },
-      noXField() {
-        return !this.fields.x_field || this.fields.x_field === "__count__";
+      hasLayers() {
+        return this.layers.length > 1;
       },
     },
 
     methods: {
+      axisOptions(layer) {
+        const dataVariable = layer.data_variable;
+        const dataOptions = this.dataOptions.find(
+          (data) => data["variable"] === dataVariable
+        );
+        return dataOptions ? dataOptions["columns"].concat("__count__") : [];
+      },
+      noColorField(layer) {
+        return !layer.color_field || layer.color_field === "__count__";
```

Comment:
We could refactor with something like:

```js
isDataField(field) {
  return !!field && field !== "__count__";
}
```

Reply:
Nice! 😃 I just think that `hasDataField` makes more sense in this case